sql/core/src/main/java/org/apache/spark/sql/sources/v2/ReadSupport.java

```diff
@@ -30,9 +30,8 @@
   /**
    * Creates a {@link DataSourceV2Reader} to scan the data from this data source.
    *
-   * @param options the options for this data source reader, which is an immutable case-insensitive
-   *                string-to-string map.
-   * @return a reader that implements the actual read logic.
+   * @param options the options for the returned data source reader, which is an immutable
+   *                case-insensitive string-to-string map.
```

Comment:
It would make this much easier to review if changes to the read path were taken out and committed in a follow-up to #19136.